### PR TITLE
FEAT: Add next-auth

### PR DESCRIPTION
**Pull request by ImPatriciaLofamia**

### CODE REVIEW

File: pages/api/auth/[...nextauth].tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+export const authOptions = {
+  // Configure one or more authentication providers
+  providers: [
+    SpotifyProvider({
+      clientId: process.env.NEXT_PUBLIC_CLIENT_ID,
+      clientSecret: process.env.NEXT_PUBLIC_CLIENT_SECRET,
+      authorization: LOGIN_URL,
+    }),
+    // ...add more providers here
```
Comment (bsistoso) [inline pages/api/auth/[...nextauth].tsx]: remove comments

File: pages/api/auth/[...nextauth].tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                refreshToken: account.refresh_token,
+                username: account.providedrAccountId,
+                accessTokenExpires: account.expires_at * 1000,
+            }
+        }
+
+        if(Date.now() < token.accessTookenExpires) {
+            return token,
+        }
+        console.log("Access token has expired, refreshing...");
```
Comment (bsistoso) [inline pages/api/auth/[...nextauth].tsx]: remove console

**Review by bsistoso — APPROVED** [review]

File: .env [hunk 3]
```diff
@@ -0,0 +1,4 @@
+NEXTAUTH_URL=http://localhost:3000
```
Comment (carlofurio-swr) [inline .env]: add space between name and value to make it more readable

File: lib/spotify.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+const scopes = [
+  "user-read-email",
+  "playlist-read-private",
+  "playlist-read-collaborative",
+  "user-read-email",
+  "streaming",
+  "user-read-private",
+  "user-library-read",
+  "user-top-read",
+  // "user-library-modify",
```
Comment (mika-ella) [inline lib/spotify.tsx]: remove if not used

**Review by carlofurio-swr — CHANGES_REQUESTED** [review]

**Review by carlofurio-swr — APPROVED** [review]

**Review by mika-ella — APPROVED** [review]

**Pull request merged**